pandas/tests/test_indexing.py

```diff
@@ -2330,6 +2330,18 @@ def test_setitem_dtype_upcast(self):
         expected = DataFrame([{"a": 1, "c" : 'foo'}, {"a": 3, "b": 2, "c" : np.nan}])
         assert_frame_equal(df,expected)
 
+        # GH10280
```

Comment:
can you add some additonal tests (these should already work), for separation of other blocks types, e.g. ``object->float``, ``float->object``).